src/sc/sidechain.cpp

```diff
@@ -739,14 +752,22 @@ void CSidechain::InitScFees()
         LogPrint("sc", "%s():%d - maxSizeOfScFeesContainers set to %d\n", __func__, __LINE__, maxSizeOfScFeesContainers);
 
         assert(scFees.empty());
+        assert(scFees_v2.empty());
 
-        Sidechain::ScFeeData defaultData(
-            lastTopQualityCertView.forwardTransferScFee, lastTopQualityCertView.mainchainBackwardTransferRequestScFee);
-        scFees.push_back(defaultData);
+        if (!isNonCeasing()) {
+            Sidechain::ScFeeData defaultData(
+                lastTopQualityCertView.forwardTransferScFee, lastTopQualityCertView.mainchainBackwardTransferRequestScFee);
+            scFees.push_back(defaultData);
```

Comment:
Prefer use `emplace_back` instead of `push_back` in cases like this.
We save a temporary object in memory and constructor is done in-place within the container.

Reply:
All occurrences of `push_back` replaced with `emplace_back` in f0bab99cd7559a9bab4c8dc65103df6eea2b4b76